### PR TITLE
Fix error/warning while building with android studio

### PR DESCRIPTION
**Pull request by volzhs**
min/target sdk should be defined in build.gradle, not in manifest.xml

updated : it's down to 16 now.
updated 2 : it's down to 1 now.

~~edit : working on other errors and warnings~~

# Functionality test
- [x] Touches
- [x] Sound
- [x] Text input
- [x] In-app purchase
- [x] Download `.obb` expansion

### CODE REVIEW

File: platform/android/AndroidManifest.xml.template [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+      xmlns:tools="http://schemas.android.com/tools"
 	  package="com.godot.game"
```
Comment (akien-mga) [inline platform/android/AndroidManifest.xml.template]: There seems to be a mixed use of tabs and spaces in this file, could you also fix that?

**Comment by akien-mga** [comment]
If some of those fixes are trivial enough, it might be worth splitting them into another PR that could be merged for 3.1, while we keep the more complex/risky ones for 3.2.

**Comment by akien-mga** [comment]
Would likely fix #6626 BTW.

**Comment by volzhs** [comment]
it's now down to 13 issues.
those seem tricky to fix

**Comment by volzhs** [comment]
oh... it's by mistake...

**Comment by volzhs** [comment]
#Weak RNG

> ../../src/com/google/android/vending/licensing/LicenseChecker.java:68: Potentially insecure random numbers on Android 4.3 and older. Read https://android-developers.blogspot.com/2013/08/some-securerandom-thoughts.html for more info.

```java
  66     private static final int TIMEOUT_MS = 10 * 1000;
  67 
  68     private static final SecureRandom RANDOM = new SecureRandom();                                  
  69     private static final boolean DEBUG_LICENSE_ERROR = false;
  70 
  71     private ILicensingService mService;
```

Now it's only 1 issue left.
it says this issue could be a problem on Android 4.3(api 18) or older.

based on https://developer.android.com/about/dashboards/
screenshot_20181218_100733
about 0.4% of devices could have a issue with it. (because we support api 18 or above)

if we must fix this issue, I would like to let someone fix it, I currenlty don't have a idea to fix it.
or we can ignore it by adding `TrulyRandom` to lint options.

@akien-mga what do you think?

**Comment by volzhs** [comment]
@akien-mga I tested most functions.
it seems pretty stable state.

File: platform/android/AndroidManifest.xml.template [hunk 2]
```diff
@@ -10,32 +11,29 @@
...
+    <service android:name="org.godotengine.godot.GodotDownloaderService" />
 
 $$ADD_APPLICATION_CHUNKS$$
 
     </application>
-    <uses-feature android:glEsVersion="0x00020000" android:required="true" />
-
-$$ADD_PERMISSION_CHUNKS$$
-
-<uses-sdk android:minSdkVersion="18" android:targetSdkVersion="27"/>
```
Comment (akien-mga) [inline platform/android/AndroidManifest.xml.template]: So we no longer define min and target SDK versions? Or is it done differently?

File: platform/android/export/export.cpp [hunk 3]
```diff
@@ -1564,7 +1564,7 @@ class EditorExportAndroid : public EditorExportPlatform {
 				_fix_resources(p_preset, data);
 			}
 
-			if (file == "res/drawable/icon.png") {
+			if (file == "res/drawable-nodpi-v4/icon.png") {
```
Comment (akien-mga) [inline platform/android/export/export.cpp]: You renamed the actual folder to `drawable-nodpi`, which is correct?

**Comment by akien-mga** [comment]
The changes to `platform/android/java/src/com/android/vending` come from upstream? Could you document the provenance and process to update them in a `platform/android/java/README.md` file?

File: platform/android/export/export.cpp (hunk 3 above)
Comment (volzhs) [reply platform/android/export/export.cpp]: yeap. I need to change this to update icon properly.
it comes with fixing `res/drawable/icon.png` warning.

File: platform/android/AndroidManifest.xml.template (hunk 2 above)
Comment (akien-mga) [reply platform/android/AndroidManifest.xml.template]: Ah I see it's done in `build.gradle` via `platform/android/SCsub`.

**Comment by volzhs** [comment]
yes. i updated it with https://github.com/google/play-licensing/tree/master/lvl_library/src/main/java/com/google/android/vending
I will add it to `READMD.md`

File: platform/android/AndroidManifest.xml.template (hunk 2 above)
Comment (volzhs) [reply platform/android/AndroidManifest.xml.template]: it should be done in build.gradle not in manifest.

File: platform/android/export/export.cpp (hunk 3 above)
Comment (volzhs) [reply platform/android/export/export.cpp]: yes. not specified dpi resource should be under `drawable-nodpi` now.

**Comment by volzhs** [comment]
@akien-mga updated.

File: platform/android/export/export.cpp (hunk 3 above)
Comment (akien-mga) [reply platform/android/export/export.cpp]: But I mean, the file is in `res/drawable-nodpi/icon.png`, but you're referencing `drawable-nodpi-v4` here, isn't this `-v4` an issue?

Comment (volzhs) [reply platform/android/export/export.cpp]: after making `.apk` for templates, it changes to `drawable-nodpi-v4`

Comment (akien-mga) [reply platform/android/export/export.cpp]: Ok, thanks for the explanation :)

File: platform/android/java/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+- Upstream: https://github.com/google/play-licensing/tree/master/lvl_library/src/main/java/com/google/android/vending
+- Version: git (eb57657, 2018) with modifications
+- License: Apache 2.0
+
+Overwrite all files under `com/google/android/vending`
+
+### Modify some files to avoid compile error and lint warning
+
+#### com/google/android/vending/licensing/util/Base64.java
+<pre><code>
```
Comment (akien-mga) [inline platform/android/java/README.md]: You can use
<pre>
```
</pre>
like on GitHub I think. Aren't `pre` + `code` redundant too?

Comment (akien-mga) [reply platform/android/java/README.md]: Maybe even
<pre>
```diff
...
```
</pre>
to get proper syntax highlighting, at least on GitHub.

Comment (volzhs) [reply platform/android/java/README.md]: I will update it soon.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**